aten/src/ATen/native/BackendSelectRegister.cpp

```diff
@@ -0,0 +1,615 @@
+    .impl_unboxedOnlyKernel<decltype(full), &full>(DispatchKey::BackendSelect)
+    .aliasAnalysis(c10::AliasAnalysisKind::FROM_SCHEMA))
+  .op(torch::RegisterOperators::options()
+    .schema("aten::full.names(int[] size, Scalar fill_value, *, Dimname[]? names, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor")
+    .impl_unboxedOnlyKernel<decltype(full_names), &full_names>(DispatchKey::BackendSelect)
+    .aliasAnalysis(c10::AliasAnalysisKind::FROM_SCHEMA))
+  .op(torch::RegisterOperators::options()
+    .schema("aten::hamming_window(int window_length, *, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor")
+    .impl_unboxedOnlyKernel<decltype(hamming_window), &hamming_window>(DispatchKey::BackendSelect)
+    .aliasAnalysis(c10::AliasAnalysisKind::FROM_SCHEMA))
+  .op(torch::RegisterOperators::options()
+    .schema("aten::hamming_window.periodic(int window_length, bool periodic, *, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor")
+    .impl_unboxedOnlyKernel<decltype(hamming_window_periodic), &hamming_window_periodic>(DispatchKey::BackendSelect)
+    .aliasAnalysis(c10::AliasAnalysisKind::FROM_SCHEMA))
+  .op(torch::RegisterOperators::options()
+    .schema("aten::hamming_window.periodic_alpha(int window_length, bool periodic, float alpha, *, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor")
+    .impl_unboxedOnlyKernel<decltype(hamming_window_periodic_alpha), &hamming_window_periodic_alpha>(DispatchKey::BackendSelect)
+    .aliasAnalysis(c10::AliasAnalysisKind::FROM_SCHEMA))
+  .op(torch::RegisterOperators::options()
+    .schema("aten::hamming_window.periodic_alpha_beta(int window_length, bool periodic, float alpha, float beta, *, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor")
+    .impl_unboxedOnlyKernel<decltype(hamming_window_periodic_alpha_beta), &hamming_window_periodic_alpha_beta>(DispatchKey::BackendSelect)
+    .aliasAnalysis(c10::AliasAnalysisKind::FROM_SCHEMA))
+  .op(torch::RegisterOperators::options()
+    .schema("aten::normal.float_float(float mean, float std, int[] size, *, Generator? generator=None, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor")
+    .impl_unboxedOnlyKernel<decltype(normal), &normal>(DispatchKey::BackendSelect)
+    .aliasAnalysis(c10::AliasAnalysisKind::FROM_SCHEMA))
+  .op(torch::RegisterOperators::options()
+    .schema("aten::triu_indices(int row, int col, int offset=0, *, ScalarType? dtype=long, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor")
+    .impl_unboxedOnlyKernel<decltype(triu_indices), &triu_indices>(DispatchKey::BackendSelect)
+    .aliasAnalysis(c10::AliasAnalysisKind::FROM_SCHEMA));
```

Comment:
IIUC this duplicates a big block of schema declarations from `native_functions.yaml`, right? If so, that seems like a pretty nontrivial downgrade to the process of adding/modifying an op, no? You have to be aware that this new slab of duplicate schema-bearing declarations exists, understand what it does

First just because it's a big redundant slab of schema-bearing declarations that have to be known to exist and manually kept in sync, second because its relationship to native_functions isn't clear - does your new op belong here or no?)

Unless I'm missing something this feels like an instance of the kind of thing we're trying to remove from the process, right? If it's really necessary, it seems like the only way to avoid the new wrinkle is to bring this code under codegen, right?

Reply:
Codegen or not, it's going to be much easier to read and/or write these if implementation and registration are colocated, e.g. 
```
Tensor full(IntArrayRef size, Scalar fill_value, const TensorOptions & options) {
  DispatchKey key = options.computeDispatchKey();
  static auto op = c10::Dispatcher::singleton().findSchemaOrThrow("aten::full", "");
  return op.callUnboxedWithDispatchKey<Tensor, IntArrayRef, Scalar, const TensorOptions &>(key, size, fill_value, options);
}

auto full_options = torch::RegisterOperators::options()
    .schema("aten::full(int[] size, Scalar fill_value, *, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor")
    .impl_unboxedOnlyKernel<decltype(full), &full>(DispatchKey::BackendSelect)
    .aliasAnalysis(c10::AliasAnalysisKind::FROM_SCHEMA);

// then later

static auto registry = torch::RegisterOperators()
  ...
  .op(full_options)
  ...

```
...or by putting an implementation lambda directly in the registry, or whatever

Reply:
...but again the larger issue is that adding these decl blocks as a second world, redundant to but untethered from native_functions.yaml, just seems like a recipe for more devex friction. Is that POV off base? 

Going back now to better understand Greg's "why do we need this at all?" comment...

Reply:
I agree on the codegen part, but @gchanan did seem too excited about this idea. it would make stuff easier to maintain and support. 

But I disagree with the redundancy of this. We have discussed this during the c10 and the follow-up meetings. we need this for dispatching, especially once TensorOptions will be gone, this code will become crucial for dispatching.

I can rewrite it following your example, no problems. We just have to agree if we want to codegen this or not.

Reply:
Part of the trouble is there is an ordering issue, where the API for registration is not as nice as it ought to be. For example, if it were possible to write registrations without having to put in the entire schema string, that would reduce the duplication quite a bit. But the work in that area isn't here yet.

Because there is an *intention* to change this API, and we may change the API multiple times, my personal thinking is that we should codegen the registrations, because it is less typing to modify a single template, than to have to modify 20 or so hand written occurrences.

Reply:
Ok, will update the PR to codegen this.

Reply:
SGTM but were you saying @gchanan wasn't digging that idea? Interested what his POV is here

Reply:
gchanan, izdeby and I had a call and gchanan is OK with the codegen, as it is simpler codegen than what we had before.